From ad0da60a380499e7b1ed8ca84ce606b183f79e39 Mon Sep 17 00:00:00 2001
From: Kevin Fenzi <kevin@scrye.com>
Date: Jul 19 2019 23:27:30 +0000
Subject: builder/kojid: revert using only 2 processors for xz compression and lower preset to 6


In c360c791ec24b567a16553dca38b24645dd80a36 we changed xz to use only
2 cpus for compression. That commit notes that xz threads are really
memory hungry. However, preset -9 is many times as memory hungry as
preset -6. If we switch to preset 6, xz compressed artifacts will be
a little bit larger, but will compress (and decompress!) a lot larger.
This is obviously a trade off, but I think lowering the preset and
using more cpus will be a better tradeoff.

This commit moves xz to preset 6 and using all cpus the builder has.

Signed-off-by: Kevin Fenzi <kevin@scrye.com>

---

diff --git a/builder/kojid b/builder/kojid
index 3f84887..09a48a9 100755
--- a/builder/kojid
+++ b/builder/kojid
@@ -3989,7 +3989,7 @@ class BaseImageTask(OzImageTask):
             'xz-cp-%s-%s.log' % (format, self.arch))
         log_output(self.session, cmd[0], cmd, conlog, self.getUploadDir(),
             logerror=1)
-        cmd = ['/usr/bin/xz', '-z9T2', rawimg]
+        cmd = ['/usr/bin/xz', '-z6T0', rawimg]
         conlog = os.path.join(self.workdir,
             'xz-%s-%s.log' % (format, self.arch))
         log_output(self.session, cmd[0], cmd, conlog, self.getUploadDir(),

